core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -212,11 +212,11 @@ abstract class RDD[T: ClassTag](
   /**
    * Mark the RDD as non-persistent, and remove all blocks for it from memory and disk.
    *
-   * @param blocking Whether to block until all blocks are deleted.
+   * @param blocking Whether to block until all blocks are deleted; default is false
    * @return This RDD.
    */
-  def unpersist(blocking: Boolean = true): this.type = {
-    logInfo("Removing RDD " + id + " from persistence list")
+  def unpersist(blocking: Boolean = false): this.type = {
+    logInfo(s"Removing RDD $id from persistence list")
```

Comment:
How about documenting this behaviour in the RDD programming guide? e.g., non-blocking by default. If you block until unpersist finishes, you need to set `blocking = true`, brabrabra...
https://spark.apache.org/docs/latest/rdd-programming-guide.html#removing-data

Reply:
Will do. We also don't have a similar note about freeing broadcasts, so I'll add a few sentences about it.